kliff/trainer/kim_trainer.py

```diff
@@ -0,0 +1,393 @@
+            self.ensure_tarball_model_installation(self.model_name, self.collection)
+
+        self.model = KIMModel(self.model_name)
+        self.parameters = self.model.get_model_params()
+
+    def setup_parameter_transforms(self):
+        """
+        This method set up the transformed parameter space for models. It can be used
+        for any model type in general, but as there exists a significant difference
+        between how models handles their parameters, it is left for the subclass to
+        implement. Although to ensure that `initialize` function remains consistent
+        this method will not raise NotImplemented error, rather it will quietly pass.
+        So be aware.
+        """
+        self.set_parameters_as_mutable()
+        mutable_params = self.model.parameters()
+        parameter_transforms_input = self.parameter_transform_options["parameter_list"]
+        if parameter_transforms_input is not None:
+            for model_params, input_params in zip(
+                mutable_params, parameter_transforms_input
+            ):
+                if isinstance(input_params, dict):
+                    param_name = list(input_params.keys())[0]
+                    if param_name != model_params.name:
+                        raise TrainerError(
+                            f"Parameter name mismatch. Expected {model_params.name}, got {param_name}."
+                        )
+
+                    param_value_dict = input_params[param_name]
+                    transform_name = param_value_dict.get("transform_name", None)
+                    params_value = param_value_dict.get("value", None)
+                    bounds = param_value_dict.get("bounds", None)
+
+                    if transform_name is not None:
+                        transform_module = getattr(
+                            importlib.import_module(
+                                f"kliff.transforms.parameter_transforms"
+                            ),
+                            transform_name,
+                        )
+                        transform_module = transform_module()
+                        model_params.add_transform(transform_module)
+
+                    if params_value is not None:
+                        model_params.copy_from_model_space(params_value)
+
+                    if bounds is not None:
+                        model_params.add_bounds_model_space(np.array(bounds))
+
+                elif isinstance(input_params, str):
+                    if input_params != model_params.name:
+                        raise TrainerError(
+                            f"Parameter name mismatch. Expected {model_params.name}, got {input_params}."
+                        )
+                else:
+                    raise TrainerError(
+                        f"Optimizable parameters must be string or value dict. Got {input_params} instead."
+                    )
+
+    def setup_optimizer(self):
```

Comment:
Should this be implemented as `get_optimizer()` below?

Reply:
setup optimizer was chosen as it might be more involved, like setting up ema, lr decay etc. I have no preference either way.